Modules/Common/SolanaLikeTokenModule.php

```diff
@@ -0,0 +1,247 @@
+<?php declare(strict_types = 1);
+
+/*  Idea (c) 2023 Nikita Zhavoronkov, nikzh@nikzh.com
+ *  Copyright (c) 2023-2024 3xpl developers, 3@3xpl.com, see CONTRIBUTORS.md
+ *  Distributed under the MIT software license, see LICENSE.md  */
+
+/**
+ * This module processes Solana SPL Token transfers.
+ * https://spl.solana.com/token
+ */
+
+abstract class SolanaLikeTokenModule extends CoreModule
+{
+    use SolanaTraits;
+
+    public ?BlockHashFormat $block_hash_format = BlockHashFormat::AlphaNumeric;
+    public ?AddressFormat $address_format = AddressFormat::AlphaNumeric;
+    public ?TransactionHashFormat $transaction_hash_format = TransactionHashFormat::AlphaNumeric;
+    public ?TransactionRenderModel $transaction_render_model = TransactionRenderModel::Mixed;
+    public ?CurrencyFormat $currency_format = CurrencyFormat::AlphaNumeric;
+    public ?FeeRenderModel $fee_render_model = FeeRenderModel::None;
+    public ?PrivacyModel $privacy_model = PrivacyModel::Transparent;
+
+    public ?array $events_table_fields = ['block', 'transaction', 'sort_key', 'time', 'address', 'currency', 'effect', 'failed'];
+    public ?array $events_table_nullable_fields = [];
+
+    public ?array $currencies_table_fields = ['id', 'name', 'symbol', 'decimals'];
+    public ?array $currencies_table_nullable_fields = [];
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = true;
+    public ?bool $allow_empty_return_events = true;
+    public ?bool $allow_empty_return_currencies = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = false;
+
+    public ?bool $ignore_sum_of_all_effects = true;
+
+    public ?array $tokens_list = null;
+
+    final public function pre_initialize()
+    {
+        $this->version = 1;
+    }
+
+    final public function post_post_initialize()
+    {
+        // For old version of SPL Token we need to get meta from token-list
+        // Ref: https://github.com/solana-labs/token-list
+        $this->tokens_list = json_decode(file_get_contents(__DIR__ . '/../SPLTokensList/solana.tokenlist.formatted.json'), true);
```

Comment:
I think it's better to `decode()` once and then `unserialize()` the data.

Reply:
Unserializing data is much faster than decoding it. Considering we'll be using this for every block (?), it's a necessary optimization.

Reply:
Is it generally possible to in our architecture to not trigger it in each block? 
I remember asking you a few times about the possibility of parsing tokens once, e.g. with `trc-10`, and you said that it's not possible because the class is initializing for every block.
So didn't get what you mean, where do you want me to put this or you just mean to keep reading the file as is and then deserialize it in traits if it's needed?